app/channel.go

```diff
@@ -1662,7 +1662,13 @@ func (a *App) PermanentDeleteChannel(channel *model.Channel) *model.AppError {
 
 // This function is intended for use from the CLI. It is not robust against people joining the channel while the move
 // is in progress, and therefore should not be used from the API without first fixing this potential race condition.
-func (a *App) MoveChannel(team *model.Team, channel *model.Channel, user *model.User) *model.AppError {
+func (a *App) MoveChannel(team *model.Team, channel *model.Channel, user *model.User, removeDeactivatedMembers bool) *model.AppError {
+	if removeDeactivatedMembers {
+		if result := <-a.Srv.Store.Channel().RemoveAllDeactivatedMembers(channel.Id); result.Err != nil {
```

Comment:
I think is not a big deal, but if something fail after this, It will end up with the deletion of memberships of deactivated users, but without the entire process succeeded.

Reply:
A pattern I've used elsewhere is for the application to manage transactions, and to pass them into store methods. With a bit of clever hackery (Postgres at least), you can have "nested transactions" (e.g. savepoints and rollbacks) within the store for when you need that granularity, but then you can solve problems like the above.

(Plus, this might speed up things like bulk import, where we can do a lot more work in a large transaction.)

Reply:
I don't think this matters enough to be worthwhile in this particular case.

Reply:
Agreed: definitely a future idea.